client/signup/wpcom-login-form/index.jsx

```diff
@@ -2,7 +2,7 @@ import config from '@automattic/calypso-config';
 import { useEffect, useRef } from 'react';
 
 function getFormAction( redirectTo ) {
-	const subdomainRegExp = /^https?:\/\/([a-z0-9-]+)\.wordpress\.com(?:$|\/)/;
+	const subdomainRegExp = /^https?:\/\/(?!public-api)([a-z0-9-]+)\.wordpress\.com(?:$|\/)/;
```

Comment:
What about writing it in a form that's more straightforward to understand?
```js
const subdomainMatch = redirectTo.match( subdomainRegExp );
if ( subdomainMatch && subdomainMatch[ 1 ] !== 'public-api' ) {
  subdomain = subdomainMatch[ 1 ] + '.';
}
```
That way, letting the original regexp match and the excluding `public-api` separately, I can understand the code even without knowing that `?!` is a negative lookahead.

Reply:
Yea, sounds good! Updated as suggested in d275b87bcec639f69933501907e0c73604878935.